walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,66 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  /**
+   * Queries a collection of available time ranges that could accommodate the meeting request
+   * @param {Collection<Event>} events The collection of events that may conflict with the reqest
+   * @param {MeetingRequest} request The request to query abailable time ranges of
+   * @return {Collection<TimeRange>} The time ranges that could accommodate the meeting request
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Initializes and gets colllections that will be used locally to query
+    List<TimeRange> blockedTimes = new ArrayList<TimeRange>();
+    Collection<TimeRange> result = new ArrayList<TimeRange>();
+    Collection<String> requestAttendees = request.getAttendees();
+
+    // Finds the set of events that block time ranges for being a valid meeting time
+    for (Event event : events) {
+      Set<String> eventAttendees = event.getAttendees();
+      boolean isBlocked = false;
+      for (String attendee : eventAttendees) {
```

Comment:
Check out the java 8 stream functions: https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html

boolean isBlocked = event.getAttendees().stream(). anyMatch(attendee -> equestAttendees.contains(attendee));

Reply:
changed to use stream. This was interesting, I didn't know this could be done in Java. thanks for the reference